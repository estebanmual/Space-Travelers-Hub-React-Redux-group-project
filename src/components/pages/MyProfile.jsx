import { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import style from './MyProfile.module.css';

function MyProfile() {
  const missions = useSelector((state) => state.missions);
  const rockets = useSelector((state) => state.rockets);
  const [reservedMissions, setReservedMissions] = useState([]);
  const [reservedRockets, setReservedRockets] = useState([]);

  useEffect(() => {
    const resMissions = missions.filter((mission) => mission.reserved && mission);
    const resRockets = rockets.filter((rocket) => rocket.reserved && rocket);
    setReservedMissions(resMissions);
    setReservedRockets(resRockets);
  }, []);

  return (
    <div className={style.container}>
      <div>
        <h1>My Missions</h1>
        <ul className={style.list}>
          {reservedMissions.map((mission) => <li key={mission.id}>{mission.name}</li>)}
        </ul>
      </div>
      <div>
        <h1>My Rockets</h1>
        <ul className={style.list}>
          {reservedRockets.map((rocket) => <li key={rocket.id}>{rocket.name}</li>)}
        </ul>
      </div>
    </div>
  );
}

export default MyProfile;
