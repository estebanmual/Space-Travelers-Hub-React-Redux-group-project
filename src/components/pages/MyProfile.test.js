import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../Redux/storeSetUp';
import MyProfile from './MyProfile';

describe('testing My Profile', () => {
  it('snapshot for the My profile ', () => {
    const tree = renderer.create(<Provider store={store}><MyProfile /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing My profile component renders correctly', () => {
    render(<Provider store={store}><MyProfile /></Provider>);
    expect(screen.getByText('My Missions')).toBeInTheDocument();
  });
});
