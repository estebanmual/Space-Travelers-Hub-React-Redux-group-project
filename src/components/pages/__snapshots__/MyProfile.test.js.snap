// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing My Profile snapshot for the My profile  1`] = `
<div
  className="container"
>
  <div>
    <h1>
      My Missions
    </h1>
    <ul
      className="list"
    />
  </div>
  <div>
    <h1>
      My Rockets
    </h1>
    <ul
      className="list"
    />
  </div>
</div>
`;
